golang/btcspv/bitcoin_spv.go

```diff
@@ -338,18 +349,20 @@ func ValidateVout(vout []byte) bool {
 
 // ExtractMerkleRootLE returns the transaction merkle root from a given block header
 // The returned merkle root is little-endian
-func ExtractMerkleRootLE(header []byte) []byte {
-	return header[36:68]
+func ExtractMerkleRootLE(header RawHeader) Hash256Digest {
+	res, _ := NewHash256Digest(header[36:68])
+	return res
 }
 
 // ExtractMerkleRootBE returns the transaction merkle root from a given block header
 // The returned merkle root is big-endian
-func ExtractMerkleRootBE(header []byte) []byte {
-	return ReverseEndianness(ExtractMerkleRootLE(header))
+func ExtractMerkleRootBE(header RawHeader) Hash256Digest {
+	LE := ExtractMerkleRootLE(header)
+	return ReverseHash256Endianness(LE)
```

Comment:
Not a big deal, but we could think about making this into a 1-liner again `return ReverseHash256Endianness(ExtractMerkleRootLE(header))`